Fix noticeable masking artifact of beatmap cards when already downloaded

Joehuu opened the pull request:
Had a similar fix before seeing https://github.com/ppy/osu/pull/20743#discussion_r994955470, but using that diff instead so co-authoring.

Not sure why it was visually left broken since. Photos are on a 1920x1080 monitor.

| Before | After |
| --- | --- |
| ![Screenshot 2024-03-13 at 11 23 00 PM](https://github.com/ppy/osu/assets/35318437/2fca359e-1ac5-448b-88fb-1875898ecd7a) | ![Screenshot 2024-03-13 at 11 21 16 PM](https://github.com/ppy/osu/assets/35318437/3e58001b-464b-45af-b88b-5854b9065ab8) |

Joehuu committed:
Fix noticeable masking artifact of beatmap cards when already downloaded

Had a similar fix before seeing https://github.com/ppy/osu/pull/20743#discussion_r994955470, but using that diff instead so co-authoring.

Co-Authored-By: Bartłomiej Dach <dach.bartlomiej@gmail.com>

peppy commented:
Please add inline explanation of what is being done and why / how this fixes the issue.

Joehuu committed:
Add comment explaining why width is limited to the button area

Same comment as https://github.com/ppy/osu/blob/a47ccb8edd2392258b6b7e176b222a9ecd511fc0/osu.Game/Screens/Select/BeatmapInfoWedgeV2.cs#L91-L92.

bdach commented:
I'm abstaining from review since that patch was originally mine and it was controversial at the time.
